packages/@aws-cdk/core/lib/private/tree-metadata.ts

```diff
@@ -49,10 +52,27 @@ export class TreeMetadata extends Construct {
       }
     });
   }
+
+  private inferAttributes(construct: IConstruct): { [key: string]: any } | undefined {
+    // check if a construct implements IInspectable
+    function canInspect(inspectable: any): inspectable is IInspectable {
+      return (inspectable as IInspectable).inspect !== undefined;
```

Comment:
This `as` is not necessary, and confuses me.

You use `as` if you already know if something is of a certain type, but here you're still testing.

Reply:
removed. interesting! I thought I could use it as a part of testing for whether it's of a certain type.